default/AI/ResourcesAI.py

```diff
@@ -29,98 +40,252 @@
+                                       "current Focus", "newFocus", "new target RP/PP")
+
+    def print_table_footer(self, priorityRatio):
+        current_industry_target = 0
+        current_research_target = 0
+        all_industry_industry_target = 0
+        all_industry_research_target = 0
+        all_research_industry_target = 0
+        all_research_research_target = 0
+        total_changed = 0
+        for pinfo in self.focus_manager.all_planet_info.values():
+            if pinfo.current_focus != pinfo.future_focus:
+                total_changed += 1
+
+            planet = pinfo.planet
+            fPP, fRP = pinfo.possible_output[pinfo.future_focus]
+            current_industry_target += fPP
+            current_research_target += fRP
+
+            iPP, iRP = pinfo.possible_output[IFocus] if IFocus in planet.availableFoci else (fPP, fRP)
+            all_industry_industry_target += iPP
+            all_industry_research_target += iRP
+
+            rPP, rRP = pinfo.possible_output[RFocus] if RFocus in planet.availableFoci else (fPP, fRP)
+            all_research_industry_target += rPP
+            all_research_research_target += rRP
+
+        print "-----------------------------------"
+        print "Planet Focus Assignments to achieve target RP/PP ratio of %.2f from current ratio of %.2f ( %.1f / %.1f )" \
+            % (priorityRatio, current_research_target / (current_industry_target + 0.0001),
+               current_research_target, current_industry_target)
+        print "Max Industry assignments would result in target RP/PP ratio of %.2f ( %.1f / %.1f )" \
+            % (all_industry_research_target / (all_industry_industry_target + 0.0001),
+               all_industry_research_target, all_industry_industry_target)
+        print "Max Research assignments would result in target RP/PP ratio of %.2f ( %.1f / %.1f )" \
+            % (all_research_research_target / (all_research_industry_target + 0.0001),
+               all_research_research_target, all_research_industry_target)
+        print "-----------------------------------"
+        print "Final Ratio Target (turn %4d) RP/PP : %.2f ( %.1f / %.1f ) after %d Focus changes" \
+            % (fo.currentTurn(), current_research_target / (current_industry_target + 0.0001),
+               current_research_target, current_industry_target, total_changed)
+
+    def print_table(self, priorityRatio):
+        """Prints a table of all of the captured sections of assignments"""
+        self.print_table_header()
+
+        for title, id_set in self.sections:
+            print Reporter.table_format % (("---------- " + title + " ------------------------------")[:33], "", "", "", "", "")
+            id_set.sort()  # pay sort cost only when printing
+            for pid in id_set:
+                pinfo = self.focus_manager.baked_planet_info[pid]
+                oldFocus = pinfo.current_focus
```

Comment:
variable naming style here and elsewhere still violates PEP8.

```oldFocus``` -> ```old_focus```

Please also check the rest of the code for this (I suppose you have some decent IDE that may check for this kind of things?)